File: src/Array/cpp/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret;
        int sum;

        for (int j = nums.size()-1; j > 0; j--) {
            sum = nums[j];
            for (int i = 0; i < j; i++) {
                sum += nums[i];
                if (sum == target) {
                    ret.push_back(i+1);
                    ret.push_back(j+1);
                }
                else {
                    sum -= nums[i];
                }
            }
        }
        return ret;
    }
};

File: src/haoxiang/HashTable/TwoSum.java
package haoxiang.HashTable;

import java.util.HashMap;

/**
 * Created by haoxiang on 16/2/11.
 */
public class TwoSum {

    public int[] twoSum(int[] nums, int target) {
        //2,7,11,15  9
        int [] res = new int [2];
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
        for(int i =0;i<nums.length;++i){
            if(map.containsKey(nums[i])){
                res[0] = map.get(nums[i]);
                res[1] = i+1;
            }else{
                map.put(nums[i],i+1);
            }

        }
            return res;
    }
}
